### PR TITLE
OKTA-527165 : Link missclick fix

### PR DESCRIPTION
**Pull request by leemchale-okta**
## Description:

This PR fixes the issue of miss clicking links due to field errors pushing the link away from the mouse cursor.

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-527165](https://oktainc.atlassian.net/browse/OKTA-527165)

### CODE REVIEW

**Comment by glenfannin-okta** [comment]
If we are going with the `mousedown` approach, there is another that needs it `TextWithHtml` component.

**Review by lesterchoi-okta — APPROVED** [review]
Is there any reasonable way to unit test this?

**Comment by leemchale-okta** [comment]
not sure if you could unit test this since jest is simulating clicks programmatically?  @glenfannin-okta what do you think?  Maybe we can just check make sure the onBlur event does not fire when mouseDown on a link?

File: src/v3/src/components/TextWithHtml/TextWithHtml.tsx [hunk 1]
```diff
@@ -32,6 +33,7 @@ const TextWithHtml: UISchemaElementComponent<{
   } = uischema.options;
   const onSubmitHandler = useOnSubmit();
 
+  const onMouseDown: ClickHandler = (e) => { e.preventDefault(); console.log('mouse down'); };
```
Comment (glenfannin-okta) [inline src/v3/src/components/TextWithHtml/TextWithHtml.tsx:36]:
```suggestion
  const onMouseDown: ClickHandler = (e) => { e.preventDefault(); };
```

**Comment by glenfannin-okta** [comment]
> not sure if you could unit test this since jest is simulating clicks programmatically? @glenfannin-okta what do you think? Maybe we can just check make sure the onBlur event does not fire when mouseDown on a link?

I think the practical test for this would be an integration test that lands on a page with the first field auto-focused, then immediately clicking a link (i.e. forgot password or sign up) that should pass without an issue. 

I dont think we have a specific test set up like this (because most of our integration tests loads the target page first - so it doesnt have a need to navigate by clicking a specific link)

**Review by glenfannin-okta — APPROVED** [review]

**Pull request merged**